src/ripple/core/DatabaseCon.h

```diff
@@ -115,8 +131,8 @@ class DatabaseCon
         std::string const& DBName,
         std::array<char const*, N> const& pragma,
         std::array<char const*, M> const& initSQL)
+        : DatabaseCon({}, dataDir / DBName, {}, pragma, initSQL)
```

Comment:
See the comment below about removing the first parameter to this function. However, _if_ we wanted to use this idiom, I'd give the full class name `DatabaseCon(Base{}, ...)` to make the code easier to navigate; I'd probably rename the `Base` parameter as well, but it's not worth bike-shedding now since I think we should remove it anyway.